src/commands/init.js

```diff
@@ -83,84 +222,323 @@ Examples:
+    return
+  }
+
+  // Fail if the output directory already exists
+  if (filesystem.exists(directory)) {
+    print.error(`Directory or file "${directory}" already exists`)
+    process.exitCode = 1
+    return
+  }
+
+  // Clone the example subgraph repository
+  let cloned = await withSpinner(
+    `Cloning example subgraph`,
+    `Failed to clone example subgraph`,
+    async spinner => {
+      await system.run(
+        `git clone http://github.com/graphprotocol/example-subgraph ${directory}`
+      )
+      return true
+    }
+  )
+  if (!cloned) {
+    process.exitCode = 1
+    return
+  }
+
+  // Update package.json to match the subgraph name
+  let prepared = await withSpinner(
+    `Update subgraph name and commands in package.json`,
+    `Failed to update subgraph name and commands in package.json`,
+    async spinner => {
+      try {
+        // Load package.json
+        let pkgJsonFilename = filesystem.path(directory, 'package.json')
+        let pkgJson = await filesystem.read(pkgJsonFilename, 'json')
+
+        pkgJson.name = getSubgraphBasename(subgraphName)
+        Object.keys(pkgJson.scripts).forEach(name => {
+          pkgJson.scripts[name] = pkgJson.scripts[name].replace('example', subgraphName)
+        })
+        delete pkgJson['license']
+        delete pkgJson['repository']
+
+        // Write package.json
+        await filesystem.write(pkgJsonFilename, pkgJson, { jsonIndent: 2 })
+        return true
+      } catch (e) {
+        print.error(`Failed to preconfigure the subgraph: ${e}`)
+        filesystem.remove(directory)
+        return false
+      }
+    }
+  )
+  if (!prepared) {
+    process.exitCode = 1
+    return
+  }
+
+  // Initialize a fresh Git repository
+  let repo = await initRepository(toolbox, directory)
+  if (repo !== true) {
+    process.exitCode = 1
+    return
+  }
+
+  // Install dependencies
+  let installed = await installDependencies(toolbox, directory, commands.install)
+  if (installed !== true) {
+    process.exitCode = 1
+    return
+  }
+
+  // Run code-generation
+  let codegen = await runCodegen(toolbox, directory, commands.codegen)
+  if (codegen !== true) {
+    process.exitCode = 1
+    return
+  }
+
+  printNextSteps(toolbox, { subgraphName, directory }, { commands })
+}
+
+const initSubgraphFromContract = async (
+  toolbox,
+  { allowSimpleName, subgraphName, directory, abi, network, address },
+  { commands }
+) => {
+  let { print } = toolbox
+
+  // Fail if the subgraph name is invalid
+  if (!revalidateSubgraphName(toolbox, subgraphName, { allowSimpleName })) {
+    process.exitCode = 1
+    return
+  }
+
+  // Fail if the output directory already exists
+  if (toolbox.filesystem.exists(directory)) {
+    print.error(`Directory or file "${directory}" already exists`)
+    process.exitCode = 1
+    return
+  }
+
+  if (abiEvents(abi).length === 0) {
+    // Fail if the ABI does not contain any events
+    print.error(`ABI does not contain any events`)
```

Comment:
should an ABI with no events fail everything? Since we'll soon get other trigger types, it may make sense to scaffold as much as we can even with no events.

Reply:
Not right now though — if we don't fail here, the subgraph would later fail to build because of there being no event handlers defined. We can improve this when the other trigger types are added.